modules/vstudio/vs2010_vcxproj.lua

```diff
@@ -3666,6 +4016,466 @@
+		-- Default to 8 parallel jobs
+		if cfg.flags.MultiProcessorCompile then
+			m.element("MultiProcNumber", nil, "8")
+		end
+	end
+
+--
+-- Android project generation functions
+--
+
+	function m.androidApplicationType(cfg)
+		m.element("Keyword", nil, "Android")
+		m.element("RootNamespace", nil, "%s", cfg.project.name)
+		if _ACTION >= "vs2019" then
+			m.element("MinimumVisualStudioVersion", nil, "16.0")
+		elseif _ACTION >= "vs2017" then
+			m.element("MinimumVisualStudioVersion", nil, "15.0")
+		elseif _ACTION >= "vs2015" then
+			m.element("MinimumVisualStudioVersion", nil, "14.0")
+		end
+		m.element("ApplicationType", nil, "Android")
+		if _ACTION >= "vs2017" then
+			m.element("ApplicationTypeRevision", nil, "3.0")
+		elseif _ACTION >= "vs2015" then
+			m.element("ApplicationTypeRevision", nil, "2.0")
+		else
+			m.element("ApplicationTypeRevision", nil, "1.0")
+		end
+	end
+
+	function m.androidProjectVersion(cfg)
+		_p(2, "<RootNamespace>%s</RootNamespace>", cfg.project.name)
+		_p(2, "<MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>")
+		_p(2, "<ProjectVersion>1.0</ProjectVersion>")
+	end
+
+	function m.androidDebugInformation(cfg)
+		if cfg.flags.Symbols then
+			m.element("GenerateDebugInformation", "true")
+		end
+	end
+
+	function m.androidDebugInformationFormat(cfg)
+		if cfg.symbols ~= "Default" then
+
+			if cfg.symbols == p.OFF then
+				m.element("DebugInformationFormat", nil, "None")
+			elseif cfg.symbols == "Full" then
+				m.element("DebugInformationFormat", nil, "FullDebug")
+			else
+				m.element("DebugInformationFormat", nil, "LineNumber")
+			end
+
+		end
+	end
+
+	function m.androidExceptionHandling(cfg, condition)
+		-- Note: Android defaults to 'off'
+		local exceptions = {
+			On = "Enabled",
+			Off = "Disabled",
+			UnwindTables = "UnwindTables",
+		}
+		if _ACTION >= "vs2015" then
+			if exceptions[cfg.exceptionhandling] ~= nil then
+				m.element("ExceptionHandling", condition, exceptions[cfg.exceptionhandling])
+			end
+		else
+			if cfg.exceptionhandling == premake.ON then
+				m.element("GccExceptionHandling", condition, "true")
+			end
+		end
+	end
+
+	function m.androidRuntimeTypeInfo(cfg, condition)
+		-- Note: Android defaults to 'off'
+		if cfg.rtti == premake.ON then
+			m.element("RuntimeTypeInfo", condition, "true")
+		end
+	end
+
+	function m.androidWarningLevel(cfg, condition)
+
+		if _ACTION >= "vs2015" then
+			if cfg.warnings and cfg.warnings ~= "Off" then
+				m.element("WarningLevel", nil, "EnableAllWarnings")
+			else
+				m.warningLevel(cfg, condition)
+			end
+		else
+			m.warningLevel(cfg, condition)
+		end
+
+	end
+
+	function m.androidClCompilePreprocessorDefinitions(cfg, condition)
+		m.preprocessorDefinitions(cfg, cfg.defines, false, condition)
+	end
+
+	function m.androidLanguageStandardCpp(cfg)
+		local cpp_langmap = {
+			["C++98"]   = "c++98",
+			["C++11"]   = "c++11",
+			["C++14"]   = "c++1y",
+			["C++17"]   = "c++1z",
```

Comment:
Isn't real/recent standard not supported (i.e c++17, c++20, ...)?

Reply:
Bear in mind a lot of this is just copy pasting the code from the original module, unaltered. There is a possibility to clean it up but it's kind of out of scope of this I think

Reply:
These aren't the values that the compiler will receive, these are the values that the UI uses. The support also goes back to VS2015, so we can't just change these values as it will break those older project files.